packages/noya-state/src/reducers/application.ts

```diff
@@ -1257,6 +1277,231 @@ export function reducer(
         else draft.selectedThemeStyleGroup = '';
       });
     }
+    case 'setTextColor': {
+      const [, color] = action;
+
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+
+      return produce(state, (draft) => {
+        accessPageLayers(draft, pageIndex, layerIndexPaths).forEach((layer) => {
+          if (layer._class !== 'text') return;
+          const encoded = layer.style?.textStyle?.encodedAttributes;
+          const attributes = layer.attributedString.attributes[0].attributes;
+
+          if (encoded) {
+            if (!encoded.MSAttributedStringColorAttribute) {
```

Comment:
Can we always assign a new color?

```ts
encoded.MSAttributedStringColorAttribute = color;
```

What does the case where we assign each r/g/b/a value individually do?

Reply:
It was to not change the swatchId  when you change the color but now I remember it should change it 😅